Make CTT dependency optional.

obilaniu opened the pull request:
# Description

Separate out ctt and speedrun into an optional feature, with its own dependencies.
The package with its new feature can be installed using

    pip install 'covid19sim[ctt]'

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

obilaniu committed:
Separate out ctt and speedrun into an optional feature, with its own
dependencies.

The package with its new feature can be installed using

    pip install 'covid19sim[ctt]'

Commit:
update README.md to include installation

pg2455 commented:
@satyaog  - Do we need to change any tests with regards to this change?

satyaog commented:
@pg2455 yes a small of change will be needed in the setup of the CI

satyaog commented on README.md:
````diff
@@ -16,11 +16,18 @@ We plan to update the simulator as more and more about COVID-19 will be known.
+To install the simulator without `ctt`
 ```
-pip install -r requirements.txt
+pip install covid19sim
````
I think for now it should read `pip install .`

pg2455 replied:
Ok! thanks for pointing that out. fixed it!

satyaog commented on README.md:
````diff
@@ -16,11 +16,18 @@ We plan to update the simulator as more and more about COVID-19 will be known.
 
+To install the simulator with [`ctt`](https://github.com/nasimrahaman/ctt)
+```
+pip install 'covid19sim[ctt]'
````
I think for now it should read `pip install '.[ctt]'`.

pg2455 committed:
Update on how to install

pg2455 commented:
@satyaog  - Can we make those changes with this PR itself?

satyaog commented:
@pg2455 , one way or the other. This PR can be merged then I’ll rebase on the new develop or merge mine then I’ll update the CI  script in this one

pg2455 commented:
@satyaog  - there is a conflict here now.

obilaniu committed:
Resolve merge conflict.

obilaniu committed:
Merge branch 'develop' into cttfix

satyaog committed:
Fix requirements

obilaniu commented:
@pg2455 Go for merge then?

satyaog committed:
Add extra require ctt-tf

This decouples tensorflow from the project allowing to run the sim in python 3.8

satyaog commented:
Btw, I added the same tweak to the ctt project to decouple the tensorflow depedency just before this got merged. The python 3.8 test is still running but should be fine
